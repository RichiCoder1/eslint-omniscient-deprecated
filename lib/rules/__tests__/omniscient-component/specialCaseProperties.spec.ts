import { fix } from "./__setup__/setup";

describe("special cased properties", () => {
    it("should handle propTypes", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", {
    propTypes: {
        label: PropTypes.string
    }
}, ({label}) => {
    return (<h1>{label}</h1>);
})`;
        const output = fix(input);
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });

    describe("defaultProps", () => {
        it("should handle default props object", () => {
            const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", {
    getDefaultProps() {
        return { label: "test" };
    }
}, ({label}) => {
    return (<h1>{label}</h1>);
})`;
            const output = fix(input);
            expect(output).toBeFixed();
            expect(output).toMatchSnapshot();
        });

        it("should handle default props method", () => {
            const input = `
import component from 'omniscient';
import { getDefaultPropsFor } from 'test';

const TestComponent = component("TestComponent", {
    getDefaultProps() {
        const test = 2;
        return getDefaultPropsFor(test);
    }
}, ({label}) => {
    return (<h1>{label}</h1>);
})`;
            const output = fix(input);
            expect(output).toBeFixed();
            expect(output).toMatchSnapshot();
        });
    });

    it("should handle initial state", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", {
    getInitialState() {
        return { i: 1 };
    }
}, ({label}) => {
    return (<h1>{label}</h1>);
})`;
        const output = fix(input);
        expect(output).toBeFixed();
        expect(output).toMatchSnapshot();
    });
});
