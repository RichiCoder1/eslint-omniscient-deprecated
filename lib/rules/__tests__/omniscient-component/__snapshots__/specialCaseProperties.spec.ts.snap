// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`special cased properties defaultProps should handle default props method 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';
import { getDefaultPropsFor } from 'test';

class TestComponent extends Component {
    static defaultProps = (() => {
        const test = 2;
        return getDefaultPropsFor(test);
    })();

    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        const { label } = this.props;
        return <h1>{label}</h1>;
    }
}",
}
`;

exports[`special cased properties defaultProps should handle default props object 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static defaultProps = { label: \\"test\\" };

    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        const { label } = this.props;
        return <h1>{label}</h1>;
    }
}",
}
`;

exports[`special cased properties should handle initial state 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    state = { i: 1 };

    render() {
        const { label } = this.props;
        return <h1>{label}</h1>;
    }
}",
}
`;

exports[`special cased properties should handle propTypes 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static propTypes = {
        label: PropTypes.string
    };

    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    render() {
        const { label } = this.props;
        return <h1>{label}</h1>;
    }
}",
}
`;
